numpy/lib/npyio.pyi

```diff
@@ -50,7 +53,17 @@ _DTypeLike = Union[
 ]
 
 class _SupportsGetItem(Protocol[_T_contra, _T_co]):
-    def __getitem__(self, key: _T_contra) -> _T_co: ...
+    def __getitem__(self, key: _T_contra, /) -> _T_co: ...
+
+class _SupportsRead(Protocol[_CharType_co]):
+    def read(self) -> _CharType_co: ...
+
+class _SupportsReadSeek(Protocol[_CharType_co]):
+    def read(self, n: int, /) -> _CharType_co: ...
+    def seek(self, offset: int, whence: int, /) -> object: ...
+
+class _SupportsWrite(Protocol[_CharType_contra]):
+    def write(self, s: _CharType_contra, /) -> object: ...
```

Comment:
The `write` output value is ignored by the likes of `np.save` and `np.savez`, so we can get away with annotating it as `object`. The same holds for a number of other functions, such as `np.fromfile`.